src/main/kotlin/uk/gov/justice/digital/hmpps/crimeportalgateway/service/SqsService.kt

```diff
@@ -1,23 +1,33 @@
 package uk.gov.justice.digital.hmpps.crimeportalgateway.service
 
 import com.amazonaws.services.sqs.AmazonSQS
+import com.amazonaws.services.sqs.model.MessageAttributeValue
 import com.amazonaws.services.sqs.model.SendMessageRequest
 import com.amazonaws.services.sqs.model.SendMessageResult
 import org.slf4j.LoggerFactory
 import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.beans.factory.annotation.Qualifier
 import org.springframework.beans.factory.annotation.Value
 import org.springframework.stereotype.Component
 
 @Component
 class SqsService(
     @Value("\${aws_sqs_queue_name:crime-portal-gateway-queue}") private val queueName: String,
-    @Autowired private val amazonSqs: AmazonSQS
+    @Autowired private val amazonSqs: AmazonSQS,
+    @Autowired @Qualifier("getOperationId") private val getOperationId: () -> String
 ) {
 
     fun enqueueMessage(externalDocumentRequest: String): String {
         val msgRequest = SendMessageRequest()
             .withQueueUrl(getQueueUrl())
             .withMessageBody(externalDocumentRequest)
+            .withMessageAttributes(
+                mapOf(
+                    "operation_Id" to MessageAttributeValue()
```

Comment:
Can this just be `"operationId" to getOperationId()` ?
I think Strings are just the default

Reply:
The only method signature that's available as far as I can see requires type MessageAttributeValue - does seem a bit clunky I'll grant you but kinda makes sense in the context of the SQS payload where it maps directly like so:
```
"MessageAttributes": {
    "operation_Id": {
        "StringValue": "e2f47e6ceb8ead83ac1499c5e6f06bd0",
        "DataType": "String"
    }
}
```